File: 2315-count-asterisks/2315-count-asterisks.cpp
class Solution {
public:
    int countAsterisks(string s) {
        int flag = 0;
        int cnt = 0;
        for (char x:s) {
            if (x == '|') {flag = 1 - flag;}
            if (flag == 0 && x == '*') {++cnt;}
        }
        return cnt;
    }
};

File: 2315-count-asterisks/2315-count-asterisks.rs
impl Solution {
    pub fn count_asterisks(s: String) -> i32 {
        s.split('|')
            .enumerate()
            .filter(|&(i, _)| i % 2 == 0)
            .flat_map(|(_, s)| s.chars())
            .filter(|&c| c == '*')
            .count() as _
    }
}